soda_machine.rb

```diff
@@ -7,12 +7,19 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    sodas.length
   end
 
   def find_soda(soda_brand)
+    sodas.find { |soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    index = sodas.find_index { |soda| soda.brand == soda_brand}
+    if index != nil
+      @cash += sodas[index].price
+      sodas.delete_at(index)
+    end
   end
```

Comment:
Looks good!  You could reuse your `find_soda` method in `sell` -- the only difference is that you'd use `#delete` instead of `#delete_at`.